presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -361,25 +362,31 @@ private static boolean process(QueryRunner queryRunner, String sql, OutputFormat
 
     private static MemoryHistory getHistory()
     {
-        MemoryHistory history;
-        File historyFile = new File(getUserHome(), ".presto_history");
+        return getHistory(new File(getUserHome(), ".presto_history"));
+    }
+
+    @VisibleForTesting
+    static MemoryHistory getHistory(File historyFile)
```

Comment:
Please change the order of commits. First extract this method, then change it.

Reply:
What's the reasoning for this? The PR is gonna be squashed-merged anyway.

Reply:
I do not see any reason why it should be merged.

Reply:
I am sorry, I meant squashed before merged :)

Reply:
Depends how this project merges PRs. Usually OSS projects squash commits when merging (to avoid endless code review commits). 
I don't like or want to re-arrange commits.

Reply:
We merge PR in way which is more logically correct.